engine/storage/configdrive/src/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilder.java

```diff
@@ -86,103 +97,156 @@ public static String buildConfigDrive(final List<String[]> vmData, final String
-            Script command = new Script("/usr/bin/genisoimage", Duration.standardSeconds(300), LOG);
-            command.add("-o", tmpIsoStore.getAbsolutePath());
-            command.add("-ldots");
-            command.add("-allow-lowercase");
-            command.add("-allow-multidot");
-            command.add("-cache-inodes"); // Enable caching inode and device numbers to find hard links to files.
-            command.add("-l");
-            command.add("-quiet");
-            command.add("-J");
-            command.add("-r");
-            command.add("-V", driveLabel);
-            command.add(tempDirName);
-            LOG.debug("Executing config drive creation command: " + command.toString());
-            String result = command.execute();
-            if (result != null) {
-                String errMsg = "Unable to create iso file: " + isoFileName + " due to " + result;
-                LOG.warn(errMsg);
-                throw new CloudRuntimeException(errMsg);
-            }
-            File tmpIsoFile = new File(tmpIsoStore.getAbsolutePath());
-            // Max allowed file size of config drive is 64MB: https://docs.openstack.org/project-install-guide/baremetal/draft/configdrive.html
-            if (tmpIsoFile.length() > (64L * 1024L * 1024L)) {
-                throw new CloudRuntimeException("Config drive file exceeds maximum allowed size of 64MB");
-            }
-            return fileToBase64String(tmpIsoFile);
+            return generateAndRetrieveIsoAsBase64Iso(isoFileName, driveLabel, tempDirName);
         } catch (IOException e) {
             throw new CloudRuntimeException("Failed due to", e);
         } finally {
-            try {
+            deleteTempDir(tempDir);
+        }
+    }
+
+    private static void deleteTempDir(Path tempDir) {
+        try {
+            if (tempDir != null) {
                 FileUtils.deleteDirectory(tempDir.toFile());
-            } catch (IOException ioe) {
-                LOG.warn("Failed to delete ConfigDrive temporary directory: " + tempDirName, ioe);
             }
+        } catch (IOException ioe) {
+            LOG.warn("Failed to delete ConfigDrive temporary directory: " + tempDir.toString(), ioe);
+        }
+    }
+
+    /**
+     *  Generates the ISO file that has the tempDir content.
+     *  We will use '/usr/bin/genisoimage' to create an ISO file based on tempDir content.
+     *
+     *  Max allowed file size of config drive is 64MB [1]. Therefore, if the ISO is bigger than that, we throw a {@link CloudRuntimeException}.
+     *  [1] https://docs.openstack.org/project-install-guide/baremetal/draft/configdrive.html
+     */
+    static String generateAndRetrieveIsoAsBase64Iso(String isoFileName, String driveLabel, String tempDirName) throws IOException {
+        File tmpIsoStore = new File(tempDirName, isoFileName);
+        Script command = new Script("/usr/bin/genisoimage", Duration.standardSeconds(300), LOG);
```

Comment:
Can you include Daan's change (or @DaanHoogland  can you push your change?) around using genisoimage or mkisofs if either is available?

Reply:
Is it in some other PR? I can take a look and include it here.

Reply:
@rafaelweingartner yes, see Daan's comment here: https://github.com/apache/cloudstack/pull/2665#issuecomment-391619983